src/llama.cpp

```diff
@@ -22107,6 +22119,8 @@ static int32_t llama_chat_apply_template_internal(
 int32_t llama_chat_apply_template(
                 const struct llama_model * model,
                               const char * tmpl,
+                              const char * prefix,
+                              const char * suffix,
```

Comment:
I don't think we need to change `llama_chat_apply_template`. This feature can be implemented outside of llama.h library

Reply:
I suppose it's better to implement it in `utils.hpp` then? A separate formatting function that is then used in `format_chat` function? Sorry for questions, I'm not quite familiar with `server`'s structure.

Reply:
Yes you can add it in `server/utils.hpp`, ideally as a branch inside `format_chat` (or can split into its own function if needed)

Reply:
Thank you, already did that. I will see if all checks are good and add it into UI.

Looks like it will have be a list of chat templates, in the end - with a choice of "custom" as this new simple solution.